library/ajax/messages/validate_messages_document_ajax.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * validate_messages_document_ajax.php is an AJAX rest api for retrieving a validation report for a given document.
+ * The user must have permissions to the specific document in order to run the validation process. This file can return
+ * results in both html or json format.
+ *
+ * @package openemr
+ * @link      http://www.open-emr.org
+ * @author    Stephen Nielson <snielson@discoverandchange.com>
+ * @copyright Copyright (c) 2022 Discover and Change, Inc. <snielson@discoverandchange.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+require_once("../../../interface/globals.php");
+require_once("$srcdir/pid.inc");
+
+use OpenEMR\Common\Acl\AclMain;
+use OpenEMR\Common\Csrf\CsrfUtils;
+use OpenEMR\Common\Twig\TwigContainer;
+use OpenEMR\Services\Cda\CdaValidateDocumentObject;
+use OpenEMR\Common\Logging\SystemLogger;
+use Document;
+
+if (!CsrfUtils::verifyCsrfToken($_GET["csrf"])) {
+    http_response_code(403);
+    CsrfUtils::csrfNotVerified(true, true, false);
```

Comment:
just making sure you are meaning to not kill the script if the csrf token check fails

Reply:
forgot to do the twig echo of the unauthorized html.  So yes, I didn't want to kill the script right there.  Wanted to echo the twig and then die.

Reply:
fixed.